File: backend/src/controllers/user_controller.py
# ====================================================================
# 🎛️ CONTROLADOR DE USUARIO
# ====================================================================
# ! Importamos librerías necesarias
from src.models.user_model import db, User
from werkzeug.security import generate_password_hash, check_password_hash

# ==============================================================
# 🆕 Crear un nuevo usuario
# ==============================================================
def create_user(data):
    hashed_password = generate_password_hash(data["password"])  # Encripta la contraseña
    new_user = User(
        nombre=data["nombre"],
        apellido=data["apellido"],
        email=data["email"],
        password=hashed_password
    )
    db.session.add(new_user)
    db.session.commit()
    return new_user.to_json()

# ==============================================================
# 🔎 Obtener un usuario por ID
# ==============================================================
def get_user_by_id(user_id):
    user = User.query.get(user_id)
    return user.to_json() if user else None

# ==============================================================
# 📋 Obtener todos los usuarios
# ==============================================================
def get_all_users():
    users = User.query.all()
    return [user.to_json() for user in users]

# ==============================================================
# ✏️ Actualizar datos de un usuario
# ==============================================================
def update_user(user_id, data):
    user = User.query.get(user_id)
    if not user:
        return None
    user.nombre = data.get("nombre", user.nombre)
    user.apellido = data.get("apellido", user.apellido)
    db.session.commit()
    return user.to_json()

# ==============================================================
# ❌ Eliminar un usuario
# ==============================================================
def delete_user(user_id):
    user = User.query.get(user_id)
    if not user:
        return None
    db.session.delete(user)
    db.session.commit()
    return {"message": "Usuario eliminado"}

File: backend/src/controllers/userController.js
//###################################################
// 🧑‍💻 Controlador de Usuarios
//###################################################

const userService = require("../services/userService");
const jwt = require("jsonwebtoken");


// ! Crear un nuevo usuario
const createUser = async (req, res) => {
  try {
    const user = await userService.createUser(req.body);

    // 🔐 Generar token JWT
    const token = jwt.sign({ id: user._id }, process.env.JWT_SECRET, { expiresIn: "7d" });
    res.status(201).json({ user, token });
  } catch (error) {
    res.status(400).json({ message: error.message });
  }
};

// ! Obtener todos los usuarios
const getUsers = async (req, res) => {
  try {
    const users = await userService.getUsers();
    res.json(users);
  } catch (error) {
    res.status(500).json({ message: "Error obteniendo los usuarios." });
  }
};

// ! Eliminar un usuario por ID
const deleteUser = async (req, res) => {
  try {
    const userId = req.params.id;
    await userService.deleteUser(userId);
    res.status(200).json({ message: "Usuario eliminado correctamente." });
  } catch (error) {
    res.status(500).json({ message: "Error eliminando el usuario." });
  }
};

// 
// Obtener un usuario por ID (si querés agregarlo después)
//  const getUserById = async (req, res) => {
//    res.status(501).json({ message: "Endpoint no implementado aún." });
//  };

// Actualizar un usuario por ID (si querés agregarlo después)
//  const updateUser = async (req, res) => {
//    res.status(501).json({ message: "Endpoint no implementado aún." });
//  };


module.exports = { getUsers, createUser, deleteUser };  //  getUserById, updateUser
